### PR TITLE
Add data to `PublicChainConfig`

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/common/config/PublicChainConfig.java [hunk 1]
```diff
@@ -31,5 +31,12 @@
 @NoArgsConstructor
 @AllArgsConstructor
 public class PublicChainConfig {
+    private Integer chainId;
+    private boolean isSidechain;
+    private String nodeAddress;
+    private String hubAddress;
     private Duration blockTime;
+    private long startBlockNumber;
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/config/PublicChainConfig.java]: I don't think we need 
```
    private long startBlockNumber;
    private float gasPriceMultiplier;
    private long gasPriceCap;
```

Comment (mcornaton) [reply src/main/java/com/iexec/common/config/PublicChainConfig.java]: Depending on what we want to achieve, either we let these values into core, or we get it from the blockchain adapter. I think it is better to have the blockchain adapter act as a configuration server and gather all the public blockchain data in this class. Don't you agree?

Comment (james-toussaint) [reply src/main/java/com/iexec/common/config/PublicChainConfig.java]: startBlockNumber is related to dealwatching, we don't plan to migrate it in short time.
gas* are here for people wanting to tune there tx costs. These values are totally up to people, we don't want to force them using remote ones.

Comment (mcornaton) [reply src/main/java/com/iexec/common/config/PublicChainConfig.java]: OK, I see. They are removed!

**Comment by james-toussaint** [comment]
PublicConfiguration should be shrinked since partial content is moved to the new class

**Comment by mcornaton** [comment]
Sure, it'll be once core PR is ready.

**Review by zguesmi — APPROVED** [review]
Thanks :+1:

**Review by james-toussaint — APPROVED** [review]
Thanks for the update!

**Pull request merged**